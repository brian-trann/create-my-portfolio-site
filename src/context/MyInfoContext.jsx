import React from 'react';
const MyInfoContext = React.createContext();
export default MyInfoContext;
