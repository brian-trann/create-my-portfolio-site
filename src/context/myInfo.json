{
    "navigation": {
        "name": "John Doe",
        "menu": [
            {
                "text": "Home",
                "url": "/",
                "target":""
            },
            {
                "text": "About",
                "url": "/about",
                "target":""
            },
            {
                "text": "Github",
                "url": "https://github.com/",
                "target":"_blank"
            },
            {
                "text": "LinkedIn",
                "url": "https://www.linkedin.com//",
                "target":"_blank"
            }
        ]
    },
    "main": {
        "greeting": "Hi, I'm John, a Software Developer from California",
        "blurb":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "profileImage": { 
            "url":"",
            "altText":"profile image"
        },
        "image": {
            "url": "/",
            "altText": "temp"
        }
    },
    "work":{
        "categoryName":"Selected Work",
        "projects":[
            {
                "projectName": "project0",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project0 thumbnail"
                }
            },
            {
                "projectName": "project1",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project1 thumbnail"
                }
            },
            {
                "projectName": "project2",
                "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
                "githubUrl":"https://github.com/",
                "projectUrl":"https://vercel.com/",
                "image": {
                    "url": "https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                    "altText": "project2 thumbnail"
                }
            }
        ],
        "buttonNames":{
            "primary":"Github",
            "secondary":"Site"
        }
    },
    "about": {
        "categoryName":"About",
        "statement": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque.",
        "skills":{
            "categoryName":"Skills",
            "items":[
                {
                    "type":"Languages",
                    "items":["JavaScript", "HTML", "CSS","Python","SQL"]
                },
                {
                    "type":"Libraries and Frameworks",
                    "items":["React", "Redux", "Express","Flask"]
                },
                {
                    "type":"Other",
                    "items":["other0", "other1"]
                }
            ]
        },
        "workExperience": {
            "categoryName":"Work Experience",
            "work": [
                {
                    "company":"company0",
                    "position":"position0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                },
                {
                    "company":"company2",
                    "position":"position1",
                    "date":"Date 2019 - Date 2020",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "education":{
            "categoryName":"Education",
            "items": [
                {
                    "name":"education0",
                    "curriculum":"curriculum0",
                    "date":"Date 2020 - Date 2021",
                    "description":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas sapien tortor, ornare nec leo nec, convallis pulvinar tortor. Ut lorem sem, tempus eu cursus eu, pulvinar sit amet neque."
                }
            ]
        },
        "certifications":{
            "categoryName":"Certifications",
            "items":["certification0", "certification1"]
        }
    },
    "other":{
        "images":[
            {"image":{
                "url":"https://github.githubassets.com/images/modules/site/home/repo-browser.png",
                "altText":"",
                "description":""
                }
            }
        ]
    }
}