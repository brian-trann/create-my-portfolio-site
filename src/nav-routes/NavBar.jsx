import React, { useContext } from 'react';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import MyInfoContext from '../context/MyInfoContext';
import { makeStyles } from '@material-ui/core/styles';
import MyNavButton from './MyNavButton';
const useStyles = makeStyles(() => ({
	root       : {
		flexGrow : 1
	},
	title      : {
		flexGrow    : 2,
		paddingLeft : '2rem'
	},
	navButtons : {
		paddingRight : '2rem'
	}
}));

const NavBar = () => {
	const { navigation: { menu, name } } = useContext(MyInfoContext);
	const classes = useStyles();
	return (
		<AppBar position='static'>
			<Toolbar>
				<Typography variant='h6' className={classes.title}>
					{name}
				</Typography>
				<div className={classes.navButtons}>
					{menu.map((e, i) => {
						return <MyNavButton key={i} to={e.url} text={e.text} target={e.target} />;
					})}
				</div>
			</Toolbar>
		</AppBar>
	);
};
export default NavBar;
