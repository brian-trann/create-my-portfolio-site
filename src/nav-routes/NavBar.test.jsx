import React from 'react';
import { render } from '@testing-library/react';
import NavBar from './NavBar';
import myInfo from '../context/myInfo.json';
import MyInfoContext from '../context/MyInfoContext';
import { MemoryRouter } from 'react-router-dom';

// smoke test
it('renders without crashing', () => {
	render(
		<MemoryRouter>
			<MyInfoContext.Provider value={myInfo}>
				<NavBar />
			</MyInfoContext.Provider>
		</MemoryRouter>
	);
});

// snapshot test
it('matches snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter>
			<MyInfoContext.Provider value={myInfo}>
				<NavBar />
			</MyInfoContext.Provider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
