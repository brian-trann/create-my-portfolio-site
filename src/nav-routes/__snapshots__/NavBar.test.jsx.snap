// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        class="MuiTypography-root makeStyles-title-5 MuiTypography-h6"
      >
        John Doe
      </h6>
      <div
        class="makeStyles-navButtons-6"
      >
        <a
          aria-current="page"
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit active"
          href="/"
          role="button"
          tabindex="0"
          target=""
        >
          <span
            class="MuiButton-label"
          >
            Home
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          href="/about"
          role="button"
          tabindex="0"
          target=""
        >
          <span
            class="MuiButton-label"
          >
            About
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiTypography-colorPrimary"
          href="https://github.com/"
          role="button"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-label"
          >
            Github
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit MuiTypography-colorPrimary"
          href="https://www.linkedin.com//"
          role="button"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiButton-label"
          >
            LinkedIn
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
